ood-portal-generator/templates/ood-portal.conf.erb

```diff
@@ -267,12 +267,23 @@ Listen <%= addr_port %>
     <%- end -%>
   </Location>
 
-  <%- if @api_cors_enabled -%>
-  # CORS setup for API requests
-  <Location "<%= @api_cors_uri %>">
+  <%- if @api_auth_enabled || @api_cors_enabled -%>
+  # Setup for API requests
+  <Location "<%= @api_uri %>">
+    RewriteEngine On
+    <%- if @api_auth_enabled -%>
+    # Client API auth check
+    RewriteCond "%{HTTP:X-OOD-API-ID}" !^[a-z0-9-]+$ [NC,OR]
+    RewriteCond "%{HTTP:X-OOD-API-CLIENT-SECRET}" !^[a-z0-9-]+$ [NC,OR]
```

Comment:
why the OR here? Wouldn't it be simpler to just remove the `!` and `OR`?

Reply:
I am not sure it will work that way. The `RewriteCond` is to return an HTTP 401 if the headers do not have a valid value and the file with the credentials hasn't been created. A valid value is an alphanumeric string + '-' symbol (to allow the use of UUIDs).

For me, the logic reads:
`IF` invalid `X-OOD-API-ID` `OR` invalid `X-OOD-API-CLIENT-SECRET` `OR` file is not there `THEN`
`RETURN` 401

Reply:
I see now. I get what you're going for, just the expression nots and ORs don't make for a straightforward expression to read.

I'm guessing this is indeed as good as it gets?

Reply:
This is the best solution that we have found that is easy to integrate and configure with how OOD works.